syntax = "proto2";

package apollo.planning;
// dp多边形路径的配置
message DpPolyPathConfig {
  optional uint32 sample_points_num_each_level = 1 [default = 9];           // 配置文件中写为7, 每个level采样7个点, 一个level是一行么?       
  optional double step_length_max = 2 [default = 15.0];                     // 步长最大为40, 最小为20
  optional double step_length_min = 3 [default = 8.0];                      // 最小为20
  optional double lateral_sample_offset = 4 [default = 0.5];                // 侧向采样误差为0.5
  optional double lateral_adjust_coeff = 5 [default = 0.5];                 // 侧向调整的参数为0.5
  // Trajectory Cost Config
  optional double eval_time_interval = 6 [default = 0.1];                   // 评估的时间间隔为0.1秒?
  optional double path_resolution = 7 [default = 0.1];                      // 路径分辨率为1.0
  optional double obstacle_ignore_distance = 8 [default = 20.0];            // 在20米之内忽略障碍物的距离
  optional double obstacle_collision_distance = 9 [default = 0.2];          // 碰撞距离为0.5米
  optional double obstacle_risk_distance = 10 [default = 2.0];              // 障碍物的风险距离为2米
  optional double obstacle_collision_cost = 11 [default = 1e3];             // 障碍物的碰撞代价为1e8
  optional double path_l_cost = 12;                                         // l的代价为6.5
  optional double path_dl_cost = 13;                                        // dl(l方向的速度)的代价为8e3
  optional double path_ddl_cost = 14;                                       // ddl(l方向的加速度)的代价为5e1
  optional double path_l_cost_param_l0 = 15;                                // 路径l的代价参数l0为1.5
  optional double path_l_cost_param_b = 16;                                 // 路径l的代价参数b为0.40
  optional double path_l_cost_param_k = 17;                                 // 路径l的代价参数k为1.5
  optional double path_out_lane_cost = 18;                                  // 输出lane的代价为1e8
  optional double path_end_l_cost = 19;                                     // 路径结束的代价为1e4
  optional double sidepass_distance = 20;                                   // 路边通过的距离为2.8米
  optional uint32 navigator_sample_num_each_level = 21;                     // 导航每个level的采样的点为3个
}
